tools/fastdds/shm.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2020 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import os
+
+if os.name == 'posix':
+    import fcntl
+elif os.name == 'nt':
+    import win32con
+    import win32file
+    import pywintypes
+    import winerror
+    import msvcrt
+
+import re
+
+class Parser:
+
+     def __init__(self, argv): 
+        parser = argparse.ArgumentParser(
+            description =  'Shared-memory commands',
+            usage = ''' shm <sub-command> [<args>]
+
+            shm sub-commands:
+
+                clean     clean SHM zombie files
+
+            ''')
+
+        parser.add_argument('command', help = 'Command to run')
+        args = parser.parse_args(argv)
+
+        if args.command == 'clean':
+            Clean().clean()
+        else:
+            print('shm: ' + args.command + ' sub-command is not valid')
+
+class Clean:
```

Comment:
I think this class could use some inline documentation here and there. Just some explanatory comments